### DISCUSSION TITLE
Implement OpenAPI (Swagger) in server endpoints with TSOA package

### DISCUSSION THREAD

**Discussion by ridly** [sveltejs/kit#9717] [Ideas]
### Describe the problem

Hello guys,

First of all, thank you for implementing this awesome framework that provides unlimited server and client application development possibilities already!

When working on a relatively bigger or more complex application there is a need to define an OpenAPI or Swagger specification somehow for the backend. As far as I know there is no such thing for SvelteKit, we can manually create validator for path params and also body validator within the endpoint.

### Describe the proposed solution

What I am suggesting is to join forces with the [TSOA team](https://github.com/lukeautry/tsoa) and somehow enable developers to use Typescript annotations to create beautiful server endpoints within SvelteKit. There could be an optional SwaggerUI endpoint too, enabled only if NODE_ENV == 'development'.

### Importance

nice to have

### Additional Information

Let's do it :smile:

**Comment by dummdidumm** [comment]
Thank you for the kind words! I don't think we should bless one specific implementation server endpoints of another. For example TRPC is another popular implementation of type safe server endpoints. Instead we should ensure that things can integrate well with SvelteKit so that other can for example create a library which integrates TSOA with SvelteKit.

I'm converting this to a discussion, maybe you'll find fellow devs who want to work on this with you.

**Comment by gil-obradors** [comment]
Hi! I'm interested in that too...
Why not something like [next-swagger-doc](https://github.com/jellydn/next-swagger-doc)... read from JSDocs ?

**Reply by BenocxX** [reply]
I would like something like that too!

**Reply by gil-obradors** [reply]
With minor changes and adaptations... this is the root project https://github.com/Surnet/swagger-jsdoc
